spec/unit/writer/test_csv.rb

```diff
@@ -0,0 +1,80 @@
+require 'spec_helper'
+module Bmg
+  module Writer
+    describe Csv do
+
+      subject{ relation.to_csv(*args) }
+
+      let(:relation) {
+        Relation.new [
+          {id: "1", name: "Bernard Lambeau"},
+          {id: "2", name: "Yoann;Guyot"}
+        ]
+      }
+
+      context 'with default options and no output' do
+        let(:args){ [] }
+
+        it 'works' do
+          expected = <<-CSV.gsub(/^\s+/, "")
+            1,Bernard Lambeau
+            2,Yoann;Guyot
+          CSV
+          expect(subject).to eql(expected)
+        end
+      end
+
+      context 'with an output only' do
+        let(:args){ [ StringIO.new ] }
+
+        it 'works' do
+          expect(subject).to eql(args[0])
+          expected = <<-CSV.gsub(/^\s+/, "")
+            1,Bernard Lambeau
+            2,Yoann;Guyot
+          CSV
+          expect(subject.string).to eql(expected)
+        end
+      end
+
+      context 'with an output, and options' do
+        let(:args){ [ {}, StringIO.new ] }
+
+        it 'works' do
+          expect(subject).to eql(args[1])
+          expected = <<-CSV.gsub(/^\s+/, "")
+            1,Bernard Lambeau
+            2,Yoann;Guyot
+          CSV
+          expect(subject.string).to eql(expected)
+        end
+      end
+
+      context 'specifying inline CSV options' do
```

Comment:
This example would be more valuable if it specified any other `col_sep` than `;`.